src/Symfony/Component/Validator/Constraints/Ip.php

```diff
@@ -31,39 +32,101 @@ class Ip extends Constraint
     public const V6 = '6';
     public const ALL = 'all';
 
+    // adds inverse FILTER_FLAG_NO_RES_RANGE and FILTER_FLAG_NO_PRIV_RANGE flags (skip both)
+    public const V4_NO_PUBLIC = '4_no_public';
+    public const V6_NO_PUBLIC = '6_no_public';
+    public const ALL_NO_PUBLIC = 'all_no_public';
+
     // adds FILTER_FLAG_NO_PRIV_RANGE flag (skip private ranges)
-    public const V4_NO_PRIV = '4_no_priv';
-    public const V6_NO_PRIV = '6_no_priv';
-    public const ALL_NO_PRIV = 'all_no_priv';
+    public const V4_NO_PRIVATE = '4_no_private';
+    public const V6_NO_PRIVATE = '6_no_private';
+    public const ALL_NO_PRIVATE = 'all_no_private';
 
     // adds FILTER_FLAG_NO_RES_RANGE flag (skip reserved ranges)
-    public const V4_NO_RES = '4_no_res';
-    public const V6_NO_RES = '6_no_res';
-    public const ALL_NO_RES = 'all_no_res';
+    public const V4_NO_RESERVED = '4_no_reserved';
+    public const V6_NO_RESERVED = '6_no_reserved';
+    public const ALL_NO_RESERVED = 'all_no_reserved';
 
     // adds FILTER_FLAG_NO_PRIV_RANGE and FILTER_FLAG_NO_RES_RANGE flags (skip both)
     public const V4_ONLY_PUBLIC = '4_public';
     public const V6_ONLY_PUBLIC = '6_public';
     public const ALL_ONLY_PUBLIC = 'all_public';
 
+    // adds inverse FILTER_FLAG_NO_PRIV_RANGE
+    public const V4_ONLY_PRIVATE = '4_private';
+    public const V6_ONLY_PRIVATE = '6_private';
+    public const ALL_ONLY_PRIVATE = 'all_private';
+
+    // adds inverse FILTER_FLAG_NO_RES_RANGE
+    public const V4_ONLY_RESERVED = '4_reserved';
+    public const V6_ONLY_RESERVED = '6_reserved';
+    public const ALL_ONLY_RESERVED = 'all_reserved';
+
     public const INVALID_IP_ERROR = 'b1b427ae-9f6f-41b0-aa9b-84511fbb3c5b';
 
+    /**
+     * @deprecated since Symfony 6.4, use const V4_NO_PRIVATE instead
```

Comment:
I wouldn't deprecate any constants: the cost of migrating to the new ones is too high IMHO compared to the benefit, which is mostly cosmetic
let's just keep aliases